hoomd/device.py

```diff
@@ -271,6 +272,29 @@ def is_available():
         """
         return _hoomd.isCUDAAvailable();
 
+    @contextlib.contextmanager
+    def enable_profiling(self):
+        """ Enable GPU profiling.
+
+        When using GPU profiling tools on HOOMD, select the option to disable
+        profiling on start. Initialize and run a simulation long enough that
+        all autotuners have completed, then open :py:func:`enable_profiling()`
+        as a context manager and continue the simulation for a time. Profiling
+        stops when the context manager closes.
+
+        Example::
+
+            with device.enable_profiling():
+                sim.run(1000)
+        """
+
+        try:
```

Comment:
Should this show a warning or other message if an exception is raised?

Reply:
The finally ensures that the profiling is stopped and the profiling tool can write the output to disk. I don't think any output other than the triggering exception itself is needed here. This is Python's recommended way to write context managers: https://docs.python.org/3.8/library/contextlib.html